lib/get-specs-from-issue.js

```diff
@@ -2,38 +2,81 @@ module.exports = getSpecsFromIssue;
 
 function getSpecsFromIssue(context, issueNumber) {
   console.log('started getting specs out of issue');
-  let issue = context.github.issues;
-  let specs = [];
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
-    context.github.issues.get(...repo, issueNumber),
-    context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
-
-    pushSpecs(specs, allCommentsBody);
-
-    return specs;
-  });
+    .then(() => Promise.all([
+      context.github.issues.get(...repo, issueNumber),
+      context.github.issues.listComments(...repo, issueNumber)
+    ]))
+    .then(data => {
+      const { issueBody, issueComments } = data;
+
+      return issueComments
+        .map(comment => comment.body)
+        .push(issueBody)
+        .join('\n');
+    })
+    .then(issueCompleteText => getSpecsArray(issueCompleteText))
+    .then(specsArray => specNamesFromMarkdownList(specsArray));
 }
 
-function pushSpecs(specs, body) {
-  body.toLowerCase();
-  let posiotionOfSpecsList = body.indexOf('## specs');
-  body = body.slice(posiotionOfSpecsList);
-  let endPosiotionOfSpecsList = body.indexOf('##');
-  body = body.slice(0, endPosiotionOfSpecsList);
-
-  let bullets = body.split('-');
-  bullets.forEach(bullet => {
-    bullet.trim();
-    if (bullet.startsWith('it')) {
-      specs.push(bullet);
+function getSpecsArray(completeText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi
+  const specsArray = [];
+  var isInsideSpecsBlock = false;
+  var specsBlockTopLevelHeader;
+
+  completeText.split('\n').forEach(line => {
+    if (!isInsideSpecsBlock) {
+      const lineMatchResult = specsBlockStartRegex.exec(line);
+      if (lineMatchResult) {
+        isInsideSpecsBlock = true;
+        specsBlockTopLevelHeader = lineMatchResult[1];
+      }
+    } else {
+      const isHigherLevelMarkdownTitle = isMarkdownTitle(line) &&
+        !isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      const isLineValid = isEmptyLine(line) ||
+        isMarkdownListItem(line) ||
+        isMarkdownTitle(line) &&
+          isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      if (isHigherLevelMarkdownTitle) {
+        isInsideSpecsBlock = false;
+      } else if (isLineValid) {
+        specsArray.push(line);
+      }
```

Comment:
This is not readable enough in my opinion. too many ifs & elses (and nested ones too!).

and the name `getSpecsArray` is not good enough IMO.
first, adding the datatype to the name should be avoided as much as possible.

and `getSpecs` is just a short version of `getSpecsFromIssue` which is your main function.

Please see the comment in my previous review and use the promise flow to call the functions as we structured them there.
With the breakdown of the functions as suggested, you should end up with no if statements or at most 1 in each function